packages/widgets-internal/components/NumberDisplay/FiatNumberDisplay.tsx

```diff
@@ -0,0 +1,100 @@
+import BigNumber from "bignumber.js";
+import { Text, useTooltip, type TextProps } from "@pancakeswap/uikit";
+import { ONE, TEN, formatNumber, formatNumberWithFullDigits } from "@pancakeswap/utils/formatNumber";
+import { type ReactNode, memo, useMemo, type ElementType, CSSProperties, useCallback } from "react";
+import { useTranslation } from "@pancakeswap/localization";
+
+export type FiatNumberDisplayProps = {
+  fiatSymbol?: string;
+  prefix?: ReactNode;
+  suffix?: ReactNode;
+  value: string | number | BigNumber;
+  maximumSignificantDigits?: number;
+  showFullDigitsTooltip?: boolean;
+  roundingMode?: BigNumber.RoundingMode;
+  as?: ElementType;
+  style?: CSSProperties;
+} & TextProps;
+
+export const FiatNumberDisplay = memo(function FiatNumberDisplay({
+  value,
+  prefix,
+  suffix,
+  maximumSignificantDigits = 8,
+  roundingMode = BigNumber.ROUND_DOWN,
+  showFullDigitsTooltip = true,
+  style,
+  fiatSymbol = "$",
+  ...props
+}: FiatNumberDisplayProps) {
+  const { t } = useTranslation();
+
+  const valueWithSymbol = useCallback((val: number | string) => `${fiatSymbol} ${val}`, [fiatSymbol]);
+
+  const valueDisplayInFullDigits = useMemo(
+    () =>
+      formatNumberWithFullDigits(value, {
+        roundingMode,
+      }),
+    [value, roundingMode]
+  );
+
+  const valueDisplay = useMemo(() => {
+    const bnValue = new BigNumber(value);
+    if (bnValue.eq(0)) {
```

Comment:
for long format and reuse logic, can we just create a utils function in pageckage/utils, maybe like `formatFiatAmount'

so that we can reuse it outside the hook

Reply:
done